src/algebra/lattice_ordered_group.lean

```diff
@@ -0,0 +1,607 @@
+$$c + (a⊓b) = (c+a)⊓(c+b).$$
+-/
+lemma add_meet_eq_add_meet_add (a b c : α) : c + (a⊓b) = (c+a)⊓(c+b) :=
+begin
+  rw le_antisymm_iff,
+  split,
+  { simp, },
+  { rw ← add_le_add_iff_left (-c), abel, rw le_inf_iff, simp, }
+end
+
+-- We use this in Bourbaki A.VI.12  Prop 9 a)
+/--
+Let `α` be a lattice ordered commutative group with identity `0` and let `a` be an element in `α`
+with negative component `a⁻`. Then
+$$a⁻ = -(a⊓0).$$
+-/
+lemma neg_eq_neg_inf_zero (a : α) : a⁻ = -(a⊓0) :=
+begin
+  unfold lattice_ordered_add_comm_group.neg,
+  rw ← neg_inj,
+  rw neg_join_eq_neg_meet_neg,
+  simp,
+end
+
+-- Bourbaki A.VI.12  Prop 9 a)
+/--
+Let `α` be a lattice ordered commutative group and let `a` be an element in `α` with positive
+component `a⁺` and negative component `a⁻`. Then `a` can be decomposed as the difference of `a⁺` and
+`a⁻`.
+-/
+lemma pos_sub_neg (a : α) : a = a⁺ - a⁻ :=
+begin
+  unfold lattice_ordered_add_comm_group.neg,
+  rw [eq_sub_iff_add_eq, add_join_eq_add_join_add, add_zero, add_right_neg, sup_comm],
+  unfold lattice_ordered_add_comm_group.pos,
+end
+
+-- Hack to work around rewrite not working if lhs is a variable
+@[nolint doc_blame_thm] lemma pos_sub_neg' (a : α) :  a⁺ - a⁻ = a := (pos_sub_neg _).symm
+
+-- Bourbaki A.VI.12  Prop 9 a)
+/--
+Let `α` be a lattice ordered commutative group and let `a` be an element in `α` with positive
+component `a⁺` and negative component `a⁻`. Then `a⁺` and `a⁻` are co-prime (and, since they are
+positive, disjoint).
+-/
+lemma pos_meet_neg_eq_zero (a : α) : a⁺⊓a⁻=0 :=
+begin
+  rw ←add_right_inj (-a⁻),
+  rw add_meet_eq_add_meet_add,
+  rw neg_add_eq_sub,
+  simp only [add_zero, add_left_neg],
+  rw pos_sub_neg',
+  rw neg_eq_neg_inf_zero,
+  simp,
+end
+
+-- Bourbaki A.VI.12 (with a and b swapped)
+/--
+Let `α` be a lattice ordered commutative group, let `a` and `b` be elements in `α`, and let
+`(a - b)⁺` be the positive componet of `a-b`. Then
+$$a⊔b = b + (a - b)⁺.$$
+-/
+lemma join_eq_add_pos_sub (a b : α) : a⊔b = b + (a - b)⁺ :=
+  calc  a⊔b = (b+(a-b))⊔(b+0) : by {rw add_zero b, rw add_sub_cancel'_right, }
+  ... = b + ((a-b)⊔0) : by { rw ← add_join_eq_add_join_add (a-b) 0 b}
+
+-- Bourbaki A.VI.12 (with a and b swapped)
+/--
+Let `α` be a lattice ordered commutative group, let `a` and `b` be elements in `α`, and let
+`(a - b)⁺` be the positive componet of `a-b`. Then
+$$a⊓b = a - (a - b)⁺.$$
+-/
+lemma meet_eq_sub_pos_sub (a b : α) : a⊓b = a - (a - b)⁺ :=
+  calc a⊓b = (a+0)⊓(a+(b-a)) : by { rw add_zero a, rw add_sub_cancel'_right, }
+  ... = a + (0⊓(b-a))        : by {rw ← add_meet_eq_add_meet_add 0 (b-a) a}
+  ... = a + ((b-a)⊓0)        : by { rw inf_comm }
+  ... = a + (-(a-b)⊓-0 )     : by { rw neg_zero, rw neg_sub }
+  ... = a - ((a-b)⊔0)        : by  { rw ← neg_join_eq_neg_meet_neg, rw ← sub_eq_add_neg, }
+
+-- Bourbaki A.VI.12 Prop 9 c)
+-- Can we rewrite `lattice_ordered_add_comm_group.le_pos b` as `b.le_pos`?
```

Comment:
No, typeclasses can't provide dot notation on the types they describe.

Reply:
Okay, thanks.